types/set/set.mojo

```diff
@@ -0,0 +1,102 @@
+from memory.unsafe import DTypePointer
+from memory import memcpy, memset_zero
+from math import abs
+
+
+struct SetInt:
+    alias resize_factor_up = 0.5
+    alias resize_factor_down = 0.25
+    alias min_size = 11
+
+    var size: Int
+    var _filled: Int
+    var data: Pointer[Int]
+    var flag_for_zero_value: Bool
+
+    fn __init__(inout self):
+        self.size = self.min_size
+        self._filled = 0
+        self.flag_for_zero_value = False
+
+        self.data = Pointer[Int].alloc(self.min_size)
+        memset_zero(self.data, self.min_size)
+
+    fn __del__(owned self):
+        self.data.free()
+
+    fn __hash(self, value: Int) -> Int:
+        return abs(value) % self.size
+
+    fn __contains(self, value: Int) -> Int:
+        if value == 0 and not self.flag_for_zero_value:
+            return -1
+
+        for i in range(self.size):
+            let index = (self.__hash(value) + i) % self.size
+            if self.data[index] == value:
+                return index
+
+        return -1
+
+    fn contains(self, value: Int) -> Bool:
+        return self.__contains(value) >= 0
+
+    fn add(inout self, value: Int):
+        if value == 0:
+            self.flag_for_zero_value = True
```

Comment:
What is wrong with adding an Int with the value 0?

Reply:
because i am using the zero values as a check to know if there already exists a value in the set or not, so because there are always zero values in the set i use a flag to know if a zero was added.